torch/quantization/fx/_equalize.py

```diff
@@ -220,3 +222,245 @@ def node_supports_equalization(node: Node, modules) -> bool:
     elif node.op == 'call_function':
         return node.target == nn.functional.linear
     return False
+
+def get_weight_eq_obs(
+    input_eq_obs_node: Node,
+    model: GraphModule,
+    modules: Dict[str, nn.Module]
+) -> Tuple[Optional[Node], Optional[_WeightEqualizationObserver]]:
+    """ Gets the following weight equalization observer. There should always
+    exsist a weight equalization observer after an input equalization observer.
+
+    Returns the node containing the weight equalization observer, and the weight
+    equalization observer if it has been newly created
+    """
+
+    # Find the next node that is either a nn.Linear or a functional layer
+    weight_node = input_eq_obs_node.next
```

Comment:
This is the actual op node, right? Should we rename it?

Reply:
This could either be an op node (in the case of `nn.Linear` layers), or the actual weight node (in the case of `nn.functional.linear` layers).

Reply:
For `F.linear` layers, why do we expect to reach the weight node starting from the input_equalization_observer?

Reply:
The way the nodes in the graph get placed is `x, inp_obs, inp_eq_obs, w, weight_quant_obs, weight_eq_obs, bias, F.linear` even though the visualized graph does not have `w -> weight_quant_obs -> weight_eq_obs` connected to `inp_eq_obs`. So if we just keep calling `node.next` then we will be able to reach the `weight_eq_obs`. I'm not sure if this is just a special case that I ran into though.

Reply:
Can we instead look at the user of the `input_eq_obs` and then determine the op node or weight node from that? Once we get the op_node we can look at its args to find the weight node.

Reply:
Yes, I think that would work! I think originally it was a little difficult because the `input/weight_eq_obs` were not necessarily always right before the op node, but with the recent changes to `prepare` we can just look at the op_node's arguments.